enginetest/queries/query_plans.go

```diff
@@ -514,17 +516,13 @@ where
 	{
 		Query: `select * from mytable alias where i = 1 and s = 'first row'`,
 		ExpectedPlan: "Filter\n" +
-			" ├─ AND\n" +
-			" │   ├─ Eq\n" +
-			" │   │   ├─ alias.i:0!null\n" +
-			" │   │   └─ 1 (tinyint)\n" +
-			" │   └─ Eq\n" +
-			" │       ├─ alias.s:1!null\n" +
-			" │       └─ first row (longtext)\n" +
+			" ├─ Eq\n" +
+			" │   ├─ alias.s:1!null\n" +
+			" │   └─ first row (longtext)\n" +
 			" └─ TableAlias(alias)\n" +
 			"     └─ IndexedTableAccess(mytable)\n" +
-			"         ├─ index: [mytable.s,mytable.i]\n" +
-			"         ├─ static: [{[first row, first row], [1, 1]}]\n" +
+			"         ├─ index: [mytable.i]\n" +
+			"         ├─ static: [{[1, 1]}]\n" +
```

Comment:
Similar question here... it looks like we aren't using an index over `mytable.s` and `mytable.i` anymore. Seems like if that index exists we should still use it.

Reply:
this index is the first strict key for the join, we pick it because it's guaranteed to return one row

Reply:
if we've found a plan that has best possible cost, we don't need to check the rest

Reply:
Make sense. Thanks for explaining!